### PR TITLE
Extend `profile-gists-link` feature

### PR DESCRIPTION
**Pull request by kidonng**
This is a continuation of https://github.com/sindresorhus/refined-github/pull/3272#issuecomment-648803294. ~~Since we are improving the profile nav, we can move non-working parts of features like `remove-projects-tab` and `hide-zero-packages` to it.~~

This PR ~~includes #3272 and also~~:

- Add counters to profile nav
- Hide empty (`0`) items

Test:

- https://github.com/kidonng (Repository/Package/Gist counts, hide project)
- https://github.com/coee (No counts)
- https://github.com/github (Organization package count)
- https://github.com/airbnb (Organization hide package)

### CODE REVIEW

**Comment by fregante** [comment]
Definitely not. Both the feature you mentioned also work on repos, not just profiles.

**Comment by kidonng** [comment]
Oh, I thought that packages are not shown on repos now.

**Comment by fregante** [comment]
Also to keep in mind: https://github.com/sindresorhus/refined-github/issues/2649

**Comment by kidonng** [comment]
Are you saying we should display "Source" (public + not fork + not archived) repo count instead? Unfortunately there is no archive state in the retuned data 😕 So I just exclude forks.

**Comment by fregante** [comment]
I think that's the problem with the API you're using. `totalCount` includes items that shouldn't be included. Use `search()` instead

File: source/features/extend-profile-nav.tsx [hunk 1]
```diff
@@ -4,7 +4,7 @@ import cache from 'webext-storage-cache';
 import select from 'select-dom';
 import elementReady from 'element-ready';
 import * as pageDetect from 'github-url-detection';
-import CodeSquareIcon from '@primer/octicons-v2/build/svg/code-square.svg';
+import CodeSquareIcon from 'octicon/code-square.svg';
```
Comment (yakov116) [inline source/features/extend-profile-nav.tsx]: Can you sort the from 
```
import CodeSquareIcon from 'octicon/code-square.svg';
import * as pageDetect from 'github-url-detection';
```

Comment (kidonng) [reply source/features/extend-profile-nav.tsx]: Shortened and sorted right before your comment

**Comment by fregante** [comment]
> I think that's the problem with the API you're using. `totalCount` includes items that shouldn't be included. Use `search()` instead

This applies to all. AFAIK closed projects are part of the `totalCount` but shouldnn't.

File: source/features/extend-profile-nav.tsx [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+		packages: user.packages.totalCount,
+		gists: user.gists.totalCount
+	};
+}, {
+	cacheKey: ([username]) => 'user-counts:' + username
+});
+
+const getOrganizationPackageCount = cache.function(async (organizationName: string): Promise<number> => {
+	const {organization} = await api.v4(`
+	  organization(login: "${organizationName}") {
```
Comment (fregante) [inline source/features/extend-profile-nav.tsx:49]: Can you merge this into the other query? `remove-projects-tab` used to query both users and organizations in one v4 call; if one failed, it was just `undefined` but it kept working.

File: source/features/extend-profile-nav.tsx [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+}, {
+	cacheKey: ([organizationName]) => 'organization-package-count:' + organizationName
+});
+
+async function extendUserNav(): Promise<void> {
+	await elementReady('.UnderlineNav-body + *');
+
+	const username = getCleanPathname();
+	const href = pageDetect.isEnterprise() ? `/gist/${username}` : `https://gist.github.com/${username}`;
+	const link = (
```
Comment (fregante) [inline source/features/extend-profile-nav.tsx:65]:
```suggestion
	const gistsLink = (
```

File: source/features/extend-profile-nav.tsx [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+			packageElement.append(<span className="Counter">{packageCount}</span>);
+		} else {
+			packageElement.remove();
+		}
+	}
+}
+
+void features.add({
+	id: __filebasename,
+	description: 'Extend profile navigation with counts, gist and hide empty items.',
```
Comment (fregante) [inline source/features/extend-profile-nav.tsx:117]:
```suggestion
	description: 'Adds counters and a link to Gist to user profiles, or hides empty sections.',
```

File: source/features/extend-profile-nav.tsx [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+	projects: number;
+	packages: number;
+	gists: number;
+}
+
+const getUserCounts = cache.function(async (username: string): Promise<UserCounts> => {
+	const {search, user} = await api.v4(`
+		search(type: REPOSITORY query: "user:${username} archived:false") {
+			repositoryCount
+		}
```
Comment (fregante) [inline source/features/extend-profile-nav.tsx:24]: Can you merge this into a single API query? 

```suggestion
		}

		organization(login: "${organizationName}") {
			packages {
				totalCount
			}
		}
```

`remove-projects-tab` used to query both users and organizations in one v4 call; if one failed, it was just `undefined` but it kept working.

**Comment by kidonng** [comment]
GitHub added repositories and packages count back:
image

**Comment by fregante** [comment]
I think it might be as is

**Comment by kidonng** [comment]
For the gist counter, I agree with what you said in https://github.com/sindresorhus/refined-github/pull/898#issuecomment-353496843, sometimes I found people prolific in gists and get many useful snippets/info from them, however without the counter it's not likely I'm going to check every user's gist page to find out that.
If you do not mind I can open a new PR to add the gist counter back.

**Comment by fregante** [comment]
>  If you do not mind I can open a new PR to add the gist counter back.

hmm why? This feature also hides empty tabs

**Comment by kidonng** [comment]
It only removes two tabs: projects on user profiles and packages on organization profiles (which needs to make a request). I don't think it's worth it. If you are interested in hiding projects on user profiles, I can probably add that in the new PR, this one has been messy.

**Comment by kidonng** [comment]
I forgot there is already `remove-projects-tab`, but the selector of the user profile is broken now, let's fix that instead.

**Pull request closed without merging**